spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/AbstractBinderTests.java

```diff
@@ -199,13 +199,13 @@ public void testSendAndReceiveNoOriginalContentType() throws Exception {
 				createConsumerProperties());
 		binderBindUnbindLatency();
 
-		Message<?> message = MessageBuilder.withPayload("foo").build();
+		Message<?> message = MessageBuilder.withPayload("foo").setHeader(MessageHeaders.CONTENT_TYPE,MimeTypeUtils.TEXT_PLAIN).build();
 		moduleOutputChannel.send(message);
 		Message<?> inbound = receive(moduleInputChannel);
 		assertThat(inbound).isNotNull();
-		assertThat(inbound.getPayload()).isEqualTo("foo");
+		assertThat(inbound.getPayload()).isEqualTo("foo".getBytes());
 		assertThat(inbound.getHeaders().get(BinderHeaders.BINDER_ORIGINAL_CONTENT_TYPE)).isNull();
```

Comment:
I guess it is good to check if `originalContentType` is null here (at least to check if the support is dropped). But, I believe we can remove the BinderHeaders.BINDER_ORIGINAL_CONTENT_TYPE reference given that we are dropping support for it.

Reply:
Also,  I still see some references of `originalContentType` especially in `AbstractMessageChannelBinder`'s `serializeAndEmbedHeadersIfApplicable` method

Reply:
I kept it for a reason, in a next PR (there's an issue for it) we will provide an interceptor that will allow users to read messages from legacy SCSt. This interceptor will need to check the presence of ORIGINAL_CONTENT_TYPE_HEADER, so there's a use for that still.